### PR TITLE
Update DG to include use cases relating to appointments

### LINKED ISSUE #51: Update Developer guide use case: Appointment and Bill related items

### PR DESCRIPTION
**Pull request by cheeggered**
Closes #51

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 (For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC11 View all appointments within a timeframe**
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:288]: @mounilsankar is there a actually situation where this will be used?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to add a specific appointment between a patient and doctor. 
+2. MediConnect adds the appointment to the list.
+
+   use case ends.
+
+**Extensions**
+
+* 1a.MediConnect detects that the arguments provided to add an appointment is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2 MediConnect requests for the correct input.
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:362]: Need to standardize whether to include this for the previous use cases

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use case resumes at step 1.
+
+* 1b. MediConnect detects that the appointment to be added already exists in the system
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:365]: @mounilsankar i dont think the we enforce uniqueness constraints based on the user guide document last time.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**MSS**
+
+1. User requests to add a specific appointment between a patient and doctor. 
+2. MediConnect adds the appointment to the list.
+
+   use case ends.
+
+**Extensions**
+
+* 1a.MediConnect detects that the arguments provided to add an appointment is invalid.
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:360]: should standardize with the previous use cases

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+      Use case ends.
+
+**Use case: UC15 Edit an appointment**
+
+**MSS**
+
+1. User requests to edit an appointment.
+2. MediConnect shows the current details of the appointment to the user.
+3. requests to edit the appointment.
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:376]: missing MediConnect? Or duplicated from step 1? Actually step 2 shouldnt be here also. Probably just request to edit then show the updated information

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-    * 3a1. AddressBook shows an error message.
+      Use case resumes at step 3.
 
+**Use case: UC16 Delete an appointment**
+
+**MSS**
+
+1. User requests to list appointments 
+2. MediConnect shows a list of appointments
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:404]: @mounilsankar should listing be part of this use case? Whats the convention for this type of use cases do you know?

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+3. User requests to delete an appointment in the list
+4. MediConnect deletes the appointment
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:412]: list empty should 1a?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   Use case ends.
+
+**Extensions**
+
+* 1a.The given command is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2 MediConnect requests for the correct input
+
+      Steps 1a1-1a2 are repeated until user inputs correct input.
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:434]: standardize to include/exclude for the other use cases

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -285,29 +285,232 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 1b. The given patient does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+      Use case ends.
+
+**Use case: UC20 Close an outstanding bill of a patient**
+
+**MSS**
+
+1. User requests to list outstanding bills a patient has
```
Comment (derekjxtan) [inline docs/DeveloperGuide.md:491]: @mounilsankar same thing should closing include listing. if so, is this supposed to refer back to the other use case instead?

**Review by derekjxtan — COMMENTED** [review]
@cheeggered some standardization issues that needs to be fixed. @mounilsankar we should discuss the other points or can just resolve if youre sure.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (mounilsankar) [reply docs/DeveloperGuide.md:362]: will look into the rest and standardise this

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (mounilsankar) [reply docs/DeveloperGuide.md:491]: Yes, should not include listibng

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (mounilsankar) [reply docs/DeveloperGuide.md:404]: It should not be part of the use case

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (mounilsankar) [reply docs/DeveloperGuide.md:365]: I think we should include it? Maybe can update the userguide as well

Comment (derekjxtan) [reply docs/DeveloperGuide.md:365]: then instead of checking whether the appointment already exists we should check for overlaps instead?

Comment (mounilsankar) [reply docs/DeveloperGuide.md:365]: yes

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (mounilsankar) [reply docs/DeveloperGuide.md:362]: Will apply this format for all invalid argument extensions. @cheeggered do take note and adjust accordingly

File: docs/DeveloperGuide.md
Comment (mounilsankar) [inline docs/DeveloperGuide.md:1]: This line "Steps 1a1-1a2 are repeated until user inputs correct input." an be removed for all the extentions.

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -288,29 +288,236 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1b. MediConnect detects that the given patient does not exist in the system.
+    * 1b1. MediConnect shows an error message.
+  
+      Use case ends.
+
+**Use case: UC19 Close an outstanding bill of a patient**
+
+**MSS**
+
+1. User requests to list outstanding bills a patient has (UC19)
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:498]: Should be linking UC17

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -288,29 +288,236 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Steps 1a1-1a2 are repeated until user inputs correct input.
+
+      Use case resumes at step 1.
+
+* 1b. MediConnect detects that the appointment to be edited does not exist in the system.
+    * 1b1. MediConnect shows an error message. 
+    * 1b2. MediConnect requests for the correct input.
+
+      Steps 1b1-1b2 are repeated until user inputs correct input.
+
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:395]: Same here as UC12

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -483,6 +483,216 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1a. The list is empty.
+
+  Use case ends.
+
+* 1b. MediConnect detects that the given index is invalid.
+    * 1b1. MediConnect shows an error message.
+    * 1b2. MediConnect requests for the correct input
+
+      Use case resumes at step 2.
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:597]: Should be resumes at step 1

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -483,6 +483,216 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+   Use case ends.
+
+**Extensions**
+
+* 1a.MediConnect detects that the given command is invalid.
+    * 1a1. MediConnect shows an error message.
+    * 1a2. MediConnect requests for the correct input
+
+      Use case resumes at step 2
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:614]: same here

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -483,6 +483,216 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:636]: Likewise here

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -483,6 +483,216 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:658]: And here

**Review by mounilsankar — APPROVED** [review]
LGTM

**Pull request merged**